### ISSUE TITLE
"Prettylights" development.

### ISSUE THREAD

**Issue by GeekinaCave** [github/pages-gem#160]
Hi, i like to know is there any beta of the "Prettylights" highlighter which can be tested.

Since the mention in a forum about include rogue in pages-gem in March, there is no more info about it.

Regards.

**Comment by Stargator** [comment]
You can also add a gem dependency for prettyhighlights in your jekyll site's GemFile without requiring github-pages to include it. Do you know of any issues github-pages currently has when prettyhighlights is the highlighter used?

**Comment by ABCDIY** [comment]
what is the web source for prettylights, because "rubygems.org" doesn´t found anything.
besides which is the web page for documentation of this gem?

**Comment by Stargator** [comment]
Well apparently, Prettylights has not been released yet. According to [this comment](https://github.com/github/pages-gem/pull/79#issuecomment-85997762) in issue #79, Prettylights is under development and once released may become Jekyll's default syntax highlighter.

> I'm 100% sure that once we release PrettyLights (our native highlighter), it will become the default in Jekyll

They seem to suggest that they are waiting to release it. If there was a beta, I think they would announce that. Prettylights will be open source, so they may have a beta for it or they may just wait until it is 1.0. Until Github announces something, we are just going to have to wait.

**Comment by geoff-codes** [comment]
@vmg Is "PrettyLights" something different than [highlights](https://github.com/atom/highlights)? You guys have _another_ syntax highlighter? Are they related?

**Comment by geoff-codes** [comment]
@vmg Never mind, I found it :smiling_imp:. But I'm still unsure why not write something like a ruby version of highlights, or do something like what Chocolat does, rather than pull the TextMate code directly. Speed? Is that worth going GPL3? I'm not gonna post it before your ready (though I'm pretty sure I could... its GPL'd after all), but some general info would be cool, if you were willing to share.

@ABCDIY The answer to your question is "yes"... but you'll have to dig for it yourself. Sorry.

**Comment by greyhwndz** [comment]
Is there an update to this guys?

**Comment by vmg** [comment]
I'm afraid we're not unable to open-source PrettyLights at this point because of licensing issues. I'm personally trying to work this out but since it's not entirely up to me, I cannot give an accurate timeframe. Sorry! :(

**Comment by geoff-codes** [comment]
> I'm afraid we're not unable to open-source PrettyLights at this point because of licensing issues.

Er, I'd say that's a pretty disingenuous statement. It's very clearly open source already. I wasn't going to divulge where I got the code from, but in light of the statement above, I find it somewhat unavoidable:

> ----------------------------------------------------------------------
>
>			LINKING EXCEPTION
>
> In addition to the permissions in the GNU General Public License,
> the author gives GitHub, Inc. unlimited permission to link the compiled
> version of this library into combinations with other programs,
> and to distribute those combinations without any restriction
> coming from the use of this file, if such distribution is performed
> by GitHub, Inc. as part of a GitHub Enterprise release.
>
> ----------------------------------------------------------------------
> 
>                     GNU GENERAL PUBLIC LICENSE
>                         Version 3, 29 June 2007
> ...

So GitHub, Inc. is not required to distribute it **separately** due to the exception. However, since you _have_ conveyed the software to anyone who downloads as the GitHub Enterprise `.ova`, its within the right of anyone who had done so to redistribute it, since you've made it available in this matter; that exception doesn't bar anyone from removing that exception and/or redistributing the code. See section 7 of the GPLv3.

To give @vmg's statement the widest interpretation I can, perhaps the issue is more accurately that GitHub _does not want_ to distribute PrettyLights as GPL3, as most (all?) of their public "open source projects" are distributed under more liberal licenses. And they're not _required_ to make it available _separately_ from GitHub Enterprise, due to the exception granted by the TextMate copyright holders.

**Comment by vmg** [comment]
> To give @vmg's statement the widest interpretation I can, perhaps the issue is more accurately that GitHub does not want to distribute PrettyLights as GPL3, as most (all?) of their public "open source projects" are distributed under more liberal licenses. And they're not required to make it available separately from GitHub Enterprise, due to the exception granted by the TextMate copyright holders.

Thanks for adding more context to the discussion. Yes, what I meant by "we are unable to open-source it at the moment" is that we don't think a *library* licensed exclusively under the GPLv3 would be of much (any?) use to the community, and we're actively trying to re-license it under MIT before we make the effort to maintain a public open-source project.

That said, the source code for the library *is* available as part of GitHub Enterprise's source code offer, although it's not particularly interesting. Most of the big optimisations we've implement we're also trying to upstream to TextMate (again, under the GPLv3 license), so I'm not sure what use could anybody have for the library unless you actually negotiate a license exception with TextMate's copyright holders, like we did.

In summary: as soon as we get permission to license this under MIT or BSD, we will build an open-source project around the library. And trust me, I really want this to happen. Until then, you can request the source code, but it will be of very little use because it's licensed under the GPLv3.

**Comment by geoff-codes** [comment]
> In summary: as soon as we get permission to license this under MIT or BSD, we will build an open-source project around the library. And trust me, I really want this to happen. Until then, you can request the source code, but it will be of very little use because it's licensed under the GPLv3.

That makes a lot of sense to me. Thanks for confirming/clearing this up, I appreciate it. :+1: :octocat:

**Comment by dchambers** [comment]
> You can also add a gem dependency for prettylights in your jekyll site's GemFile without requiring github-pages to include it. Do you know of any issues github-pages currently has when prettylights is the highlighter used?

@Stargator, are you suggesting that it's possible to confingure GitHub Pages to use PrettyLights instead of Pygments? If so, are you able to share some config?

**Comment by GeekinaCave** [comment]
Well people, officially Jekyll 3.0.0 is out.
Github Pages & Prettylights, your move.

**Comment by Stargator** [comment]
@dchambers If you have the prettylights gem, then I would say yes. But since Prettylights is not readily available, then I think it would be best to wait until GitHub is able to release it under a different license.

I had no prior experience/knowledge of prettylights before this issue was opened.

**Comment by dchambers** [comment]
> @dchambers If you have the prettylights gem, then I would say yes. But since Prettylights is not readily available, then I think it would be best to wait until GitHub is able to release it under a different license.

Ah, okay, I thought it was possible that this gem was available to processes being run on GitHub Pages, even though the gem wasn't publicly available. That being the case, I wanted to know the config needed to use it!

**Comment by benbalter** [comment]
Via @vmg:

> I'm afraid we're not unable to open-source PrettyLights at this point because of licensing issues. I'm personally trying to work this out but since it's not entirely up to me, I cannot give an accurate timeframe. 

@GeekinaCave thanks for opening the issue. Closing for now, until things change.

**Comment by cup** [comment]
https://github.com/github/markup/issues/1266
